### DISCUSSION TITLE
Support blockquotes

### DISCUSSION THREAD

**Discussion by jaap-karssenberg** [zim-desktop-wiki/zim-desktop-wiki#1671] [General]
Support blockquotes to mark sections of text that are a quotation

- source: prefix line with one or more ">" (email convention)
- rendering: show one or more lines in the left margin
- implementation: embedded pageview

Important to test is that the quote can contain any markup, including headings, lists, images. By using an embedded pageview object, all features are supported, but nested.

Depends on making embedded pageviews work in the first place. This means menubar actions are directed to the view that has focus and signals are propagated. Probably needs some kind of facade design where a facade manages multiple nested buffers.

**Comment by dgasaway** [comment]
Would it be feasible to start with a "simple" version that cannot support nested formats?

**Comment by Notsonoble** [comment]
">" is also the markdown format, and it supports nesting. 

Maybe it should work like the table or code plugin?

**Comment by rscmbbng** [comment]
blockquote would be very useful indeed

**Comment by jaap-karssenberg** [comment]
Realized that markdown supports blockquotes, but not indented paragraphs in the way zim has them. Thinking about whether it makes sense to render blockquotes as indented paragraphs and vice versa. In that scenario there would be no specific support for blockquotes, but markdown blockquotes would be shown as indented paragraphs.

**Comment by radupotop** [comment]
Hi @jaap-karssenberg I wanted to propose the same enhancement and just found this discussion topic already opened.

Blockquotes would definitely be useful, I find myself using indent + italics to achieve the same visual effect. But of course it doesn't have the same semantic meaning.

Is there any movement at all on this? Is there anything blocking it?  If I wanted to implement this, can you recommend where to start? 

Thanks!

**Comment by jaap-karssenberg** [comment]
My current take on it is that we should simply map block quotes to indent
in zim and vice versa. The rationale is that otherwise when we want to
support markdown there is nothing to map indented paragraphs to in markdown.

Probably this should together with the option to modify the style - e.g.
make indented paragraphs italic by default.

And it works best if we then also support paragraphs in list items, so we
can differentiate between indented paragraphs that behave as block quotes
and indented paragraphs that are part of a list.

Afraid I don't see any easy things to speed this up - still need to some
refactoring before starting on markdown integration and my planning for Zim
is notoriously unpredictable :/

That being said, there could still be plugins that add nested text buffers
for special cases - much like the code block plugin. Today you can easily
make such a plugin, as long as you do not require formatting in the buffer.
To make formatting work, there is further work needed on the PageView to
connect formatting actions to nested buffers and have awareness of where
the cursor / focus is. Also non-trivial, but I can make a design sketch if
it is something you would like to try.

--- Jaap
